### PR TITLE
GreenField: Add Server Info API basics

### LINKED ISSUE #1534: GreenField API: GET Server Info
` GET /api/v1/server/info`, accessable if just logged in, no specific permissions.
```json
{
   "status": {
     "fullySynched": true,
     "syncStatus": [
        {
            "cryptoCode": "BTC",
            "blockHeaders": 1029392,
            "progress": 0.99,
        }
      ]
   },
   "version": "1.0.4.1",
   "onion": "blah.onion",
   "supportedPaymentMethods": [ "BTC_OnChain", "BTC_OffChain" ]
}
```

_Originally posted by @NicolasDorier in https://github.com/btcpayserver/btcpayserver/pull/1515#issuecomment-620659473_

### PR DESCRIPTION
**Pull request by dennisreimann**
Any kind of feedback welcome, as I'm still trying to learn how C# and the rest works ;)

When merged it closes #1534. 

This is what the output looks like:

```json
{
    "status": {
        "fullySynched": true,
        "syncStatus": [
            {
                "cryptoCode": "LBTC",
                "blockHeaders": 390,
                "progress": 1.0
            },
            {
                "cryptoCode": "LTC",
                "blockHeaders": 272,
                "progress": 1.0
            },
            {
                "cryptoCode": "BTC",
                "blockHeaders": 898,
                "progress": 1.0
            }
        ]
    },
    "version": "1.0.4.4",
    "onion": "http://wsaxew3qa5ljfuenfebmaf3m5ykgatct3p6zjrqwoouj3foererde3id.onion",
    "supportedPaymentMethods": [
        "BTC",
        "LBTC",
        "USDT",
        "ETB",
        "LCAD",
        "LTC",
        "BTC_LightningLike",
        "LBTC_LightningLike",
        "LTC_LightningLike"
    ]
}
```

### CODE REVIEW

File: BTCPayServer/BTCPayServer.csproj [hunk 1]
```diff
@@ -21,6 +21,10 @@
     <None Remove="Build\**" />
     <None Remove="wwwroot\bundles\jqueryvalidate\**" />
     <None Remove="wwwroot\vendor\jquery-nice-select\**" />
+    <Content Update="wwwroot\swagger\v1\swagger.template.serverinfo.json">
```
Comment (Kukks) [inline BTCPayServer/BTCPayServer.csproj:24]: Remove this

File: BTCPayServer/Controllers/GreenField/ServerInfoController.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        
+        [Authorize(AuthenticationSchemes = AuthenticationSchemes.Greenfield)]
+        [HttpGet("~/api/v1/server/info")]
+        public async Task<ActionResult> ServerInfo()
+        {
+            var stores = await _storeRepository.GetStoresByUserId(_userManager.GetUserId(User));
+            var supportedPaymentMethods = _paymentMethodHandlerDictionary
+                .SelectMany(handler => handler.GetSupportedPaymentMethods().Select(id => id.ToString()))
+                .Distinct();
+            var syncStatus = _dashBoard.GetAll()
```
Comment (Kukks) [inline BTCPayServer/Controllers/GreenField/ServerInfoController.cs:50]:
```suggestion
            var syncStatus = _dashBoard.GetAll().Where(summary => summary.Network.ShowSyncSummary)
```

File: BTCPayServer/Controllers/GreenField/ServerInfoController.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                .Where(summary => summary.Network.ShowSyncSummary)
+                .Select(summary => new ServerInfoSyncStatusData
+                {
+                    CryptoCode = summary.Network.CryptoCode,
+                    BlockHeaders = summary.Status.ChainHeight,
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/GreenField/ServerInfoController.cs:55]: `summary.Status` can actually be null, so this would crash.

**Comment by NicolasDorier** [comment]
merging now, but will fix some potential crashes.

**Pull request merged**